frontend/src/metabase/admin/permissions/selectors/data-permissions/permission-editor.ts

```diff
@@ -154,29 +154,31 @@ export const getDatabasesPermissionEditor = createSelector(
     let entities: any = [];
 
     if (schemaName != null || hasSingleSchema) {
-      const schema: Schema = hasSingleSchema
+      const schema = hasSingleSchema
         ? metadata?.database(databaseId)?.getSchemas()[0]
         : metadata?.database(databaseId)?.schema(schemaName);
 
-      entities = schema
-        .getTables()
-        .sort((a, b) => a.display_name.localeCompare(b.display_name))
-        .map(table => {
-          const entityId = getTableEntityId(table);
-          return {
-            id: table.id,
-            name: table.display_name,
-            entityId,
-            permissions: buildFieldsPermissions(
+      if (schema) {
```

Comment:
This change I'm not 100% certain about. The `schema` value here is potentially `undefined`, so is the answer to leave entities as `[]`?